x-pack/plugins/fleet/public/components/agent_enrollment_flyout/steps.tsx

```diff
@@ -5,69 +5,190 @@
  * 2.0.
  */
 
-import React, { useCallback, useMemo, useState } from 'react';
-import { EuiText, EuiButton, EuiSpacer } from '@elastic/eui';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  EuiText,
+  EuiButton,
+  EuiSpacer,
+  EuiCode,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiCopy,
+  EuiCodeBlock,
+} from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n-react';
 import { i18n } from '@kbn/i18n';
 import semverMajor from 'semver/functions/major';
 import semverMinor from 'semver/functions/minor';
 import semverPatch from 'semver/functions/patch';
 
 import type { AgentPolicy } from '../../types';
-import { useKibanaVersion } from '../../hooks';
+import { useGetSettings, useKibanaVersion, useStartServices } from '../../hooks';
+
+import { agentPolicyRouteService } from '../../../common';
+
+import { sendGetK8sManifest } from '../../hooks/use_request/k8s';
 
 import { AdvancedAgentAuthenticationSettings } from './advanced_agent_authentication_settings';
 import { SelectCreateAgentPolicy } from './agent_policy_select_create';
 
-export const DownloadStep = (hasFleetServer: boolean) => {
+export const DownloadStep = (hasFleetServer: boolean, isK8s: string, enrollmentAPIKey: string) => {
```

Comment:
you could make `isK8s` and `enrollmentAPIKey` optional parameters, to avoid passing `''` when not needed